forem/forem
# Add skip links to readinglist, podcasts, and listings

## What type of PR is this? (check all applicable)


## Description

This PR adds basic skip link functionality to pages:

- `/pod`
- `/listings`
- `/readinglist`

NB: In the current implementation, the skip link is only the first focusable element on a fresh page load and not on internal navigation. At the moment, if you navigate from home to e.g. reading list, the first focusable element will be after the header bar.

As part of #1153 we will want to change that, but I think it should be completed last, after the individual pages have the skip link functionality set up. Otherwise we'll be degrading the UX in the interim, with users having to tab through the whole header on each page.

## Related Tickets & Documents

Contributes towards #1153 

## QA Instructions, Screenshots, Recordings


https://user-images.githubusercontent.com/20773163/108725551-69d57e80-751e-11eb-90b8-955f334f8fec.mp4


For each page noted above:
- Visit the URL (fresh page load, not internal navigation)
- Press the `tab` key - the skip link should be visible
- Press enter on the keyboard - you should be scrolled to the main content on the page
- Pressing `tab` past the skip link, or clicking/pressing enter on it should result in the skip link being hidden again
- Clicking on the skip link (when visible) should have the same effect as pressing enter

### UI accessibility concerns?

This introduces the skip-link accessibility feature to the pages noted above. No additional test notes needed other than those in the previous section.

## Added tests?

- [X] No, and this is why: Updated existing tests affected by the change, not able to test the actual skip link behaviour with Cypress yet due to `tab` event not being supported

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_


## [optional] What gif best describes this PR or how it makes you feel?

![a large mascot jumps rope](https://media1.giphy.com/media/LVWzUCdGioDBe/giphy.gif?cid=ecf05e47al73vouq2p29ebdgtr4blzj75zk3g5g5vtz6e4su&rid=giphy.gif)
